controllers/constants.go

```diff
@@ -1,10 +1,12 @@
 package controllers
 
 const (
-	etcdClientPort    = 2379
-	etcdPeerPort      = 2380
-	etcdDataMountPath = "/var/lib/etcd"
-	appName           = "etcd"
-	appLabel          = "app.kubernetes.io/name"
-	clusterLabel      = "etcd.improbable.io/cluster-name"
+	etcdClientPort          = 2379
+	etcdPeerPort            = 2380
+	etcdDataMountPath       = "/var/lib/etcd"
+	appName                 = "etcd"
+	appLabel                = "app.kubernetes.io/name"
+	clusterLabel            = "etcd.improbable.io/cluster-name"
+	scheduleLabel           = "etcd.improbable.io/backup-schedule-name"
+	scheduleCancelFinalizer = "finalizers.etcd.improbable.io/backup-schedule-cancel"
```

Comment:
Instead of using a finalizer here to 'deschedule' things, can we instead have a controller that 'maintains' the CronSchedule structure via the watch on the API? i.e. schedule something to happen in N minutes on Add or de-schedule it on Delete?

Reply:
can you elaborate a bit? that sounds like what we're currently doing, but using the finalizer to safely handle the deletion